constraints/recursiveCatEvals.js

```diff
@@ -35,29 +35,47 @@ function isMaximal(parent, child){
 }
 
 /* Function that takes a tree and the category of its root's parent node
-	and labels all the nodes in the tree as being minimal or maximal 
+	and labels all the nodes in the tree as being minimal or maximal
 	instance of whatever category k they are, where:
 	minimal = does not dominate any nodes of category k
 	maximal = is not dominated by any nodes of category k
-	and level ordering is assumed (a node of category level k 
+	and level ordering is assumed (a node of category level k
 	will never be dominated by a node of category < k).
-	
-	Previous isMin or isMax labels are preserved.
+
+	When calling markMinMax in a recursive function, call without parcat arg! -MT
+
+	**Previous isMin or isMax labels are NOT preserved as of 7/27/19 -MT**
 */
 // Move this to the prosodic hierarchy file probably?
 var sCat = ["cp", "xp", "x0"];
 
 function markMinMax(mytree, parcat){
+	//on first call, initialize mytree.parentCat
+	if(parcat == void(0)){
+		mytree.parentCat = '';
+	}
+
+	// Check if node is being reused by GEN
+	if(parcat && mytree.parentCat !== parcat){
+		//console.log(mytree.id + ' is being reset');
+		//node is being reused and must be reset because it has incorrect data
+		mytree.parentCat = parcat; //correct parentCat
+		mytree.isMax = void(0); //set to undefined
+		mytree.isMin = void(0); //set to undefined
+	}
+
 	// Check for maximalitys
-	if(!mytree.hasOwnProperty('isMax')){
-		mytree.isMax = (mytree.cat !== parcat)
+	//mytree.isMax might be undefined (void(0)) as a result of if statement
+	if(!mytree.hasOwnProperty('isMax') || mytree.isMax === void(0)){
```

Comment:
I don't think we need this if conditional at all.

Reply:
once the property is initialized, hasOwnProperty() will return true even if the property is set to void(0).

Reply:
issue resolved in latest commit